internal/partinfo/partinfo.go

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2023 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package partinfo
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+)
+
+type FSType string
+
+const (
+	FAT32 FSType = "fat32"
+	Ext   FSType = "ext"
+)
+
+type PartInfo interface {
+	GetDevNode() (string, error)
+	GetLabel() (string, error)
+	GetFSType() (FSType, error)
+}
+
+type partition struct {
+	path   string
+	label  string
+	fsType FSType
+}
+
+func (b *partition) parseSuperblock() error {
+	f, err := os.OpenFile(b.path, os.O_RDONLY, 0)
+	if err != nil {
+		return fmt.Errorf("cannot parse superblock: %w", err)
```

Comment:
Once again, you don't need to prefix all of these with `cannot parse superblock` -- the is normally done by the caller. However, here you would/could provide the context of the operation you're performing, for example `fmt.Errorf("cannot open superblock: %w", err)`.